### ISSUE TITLE
Obfuscated app is not working

### ISSUE THREAD

**Issue by GaborPeto** [braintree/braintree_android#444]
### General information

* SDK/Library version: 4.5.0
* Environment: Sandbox
* Android Version and Device: Pixel 4a, Android 11
* Braintree dependencies:
        `com.braintreepayments.api:three-d-secure:4.5.0.brainTreePayments`

### Issue description
`ThreeDSecureClient.performVerification` fails when `minifyEnabled true` is set in gradle config.
The debug build works as expected, however our obfuscated (R8) version is failing.

```
        val request = ThreeDSecureRequest().apply {
            this.nonce = nonce
            this.amount = amount
            this.versionRequested = ThreeDSecureRequest.VERSION_2
        }

        threeDSecureClient.performVerification(activity, request) { lookupResult, lookupError ->
            //Here lookupError is not null!
        }
```

The error thrown is:

```
org.json.JSONException: JSONObject["pareq"] not a string.
at s.d.b.getString(SourceFile:840)
at h.i.a.s1.b(SourceFile:52)
at h.i.a.v1.b(SourceFile:53)
at h.i.a.q1$d.a(SourceFile:409)
at h.i.a.v0$b.run(SourceFile:123)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7664)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

There is nothing specific for braintree in our R8 config file as per the [Braintree documentation](https://developer.paypal.com/braintree/docs/guides/client-sdk/setup/android/v4):
```
A ProGuard configuration is provided as part of the Braintree Android SDK. There is no need to add any Braintree specific rules to your ProGuard configuration.
```

**Edit:**

It's worth noting that in our sandbox environment when we try to pay with a card that will be challenged (e.g. 4111111111111111)  the Braintree SDK is working as expected and is not throwing the exception. It's only when we try to pay with a "No-Challenge" card (e.g. 4000000000001000) we get the exception if the app code is obfuscated.

**Comment by sarahkoop** [comment]
Hi @GaborPeto - Thanks for using the Braintree Android SDK. We are looking into this issue and will post an update here when we have one.

**Comment by GaborPeto** [comment]
Thank you @sarahkoop 
I added some additional information to the issue.
I hope that will help the resolution.

**Comment by sarahkoop** [comment]
Hi @GaborPeto - After some investigation, this could actually be related to the `pareq` value being null. We've implemented a fix for this in #448 and will post an update here when it's released.

**Comment by sarahkoop** [comment]
Hi @GaborPeto - the change mentioned above was released in `4.7.0`. Can you confirm if upgrading resolves this issue?

**Comment by sarahkoop** [comment]
Closing this as resolved, but please feel free to re-open if upgrading to `4.7.0` does not resolve your issue.
